eggbox_potential_sampler/eggbox_pes_data_source/tests/test_data_source.py

```diff
@@ -0,0 +1,113 @@
+import unittest
+
+from unittest import mock
+
+import numpy as np
+from traits.testing.api import UnittestTools
+from force_bdss.api import DataValue, Slot, BaseDataSourceFactory
+
+from eggbox_potential_sampler.eggbox_pes_data_source.data_source\
+    import EggboxPESDataSource
+
+from eggbox_potential_sampler.eggbox_pes_data_source.model\
+    import EggboxPESDataSourceModel
+
+
+class TestEggboxPESDataSource(unittest.TestCase, UnittestTools):
+    def setUp(self):
+        super(TestEggboxPESDataSource, self).setUp()
+        self.factory = mock.Mock(spec=BaseDataSourceFactory)
+
+    def test_initialization(self):
+        ds = EggboxPESDataSource(self.factory)
+        self.assertEqual(ds.factory, self.factory)
+
+    def test_1d_run(self):
+        ds = EggboxPESDataSource(self.factory)
+        model = EggboxPESDataSourceModel(self.factory)
+        model.num_cells = 5
+        model.dimension = 1
+        model.sigma_star = 0.1
+
+        i = 0
+        while i < 10:
+            random = np.random.rand()
```

Comment:
I think as a general rule it's good to avoid randomness in tests so you don't get intermittent test failures. I get that you may be able to identify more corner cases by sampling a wider space, but ideally these corner cases would be something you test for directly

Reply:
I guess on the back of that comment, running the test multiple times may not be needed if you choose to go with a non-random value